cpp/src/arrow/array/array_binary.cc

```diff
@@ -26,6 +26,7 @@
 #include "arrow/type_traits.h"
 #include "arrow/util/checked_cast.h"
 #include "arrow/util/logging.h"
+#include "arrow/util/utf8.h"
```

Comment:
Is this header necessary? Looks this file calls `internal::ValidateUTF8`, not `util::ValidateUTF8`.

Reply:
`ValidateUTF8` is part of `utf8.h`, the inlined version is named `ValidateUTF8Inline`.

Reply:
`ValidateUTF8` in array/validate.h?
https://github.com/apache/arrow/blob/master/cpp/src/arrow/array/validate.h#L53

Reply:
Ah! Sorry, I might have misread then.